### PR TITLE
Dagger Implementation Attempt 1 FA-35

### PR DESCRIPTION
**Pull request by brosscode**

### CODE REVIEW

File: app/src/main/java/com/example/drawingactivity/DrawActivity.kt [hunk 1]
```diff
@@ -30,9 +32,14 @@ class DrawActivity : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        val viewModel: DrawViewModel by viewModels {
```
Comment (ForzenAcademy) [inline app/src/main/java/com/example/drawingactivity/DrawActivity.kt:35]: Should be global

File: app/src/main/java/com/example/drawingactivity/DrawActivity.kt [hunk 2]
```diff
@@ -30,9 +32,14 @@ class DrawActivity : AppCompatActivity() {
...
+            MyViewModelFactory(this, savedInstanceState)
+        }
         val binding = ActivityMainBinding.inflate(layoutInflater)
         val view = binding.root
         setContentView(view)
+        (application as MyApplication).applicationComponent.inject(this)
```
Comment (ForzenAcademy) [inline app/src/main/java/com/example/drawingactivity/DrawActivity.kt:41]: Should have activity component

**Review by ForzenAcademy — APPROVED** [review]

File: app/src/main/java/com/example/drawingactivity/DrawActivity.kt (hunk 1 above)
Comment (brosscode) [reply app/src/main/java/com/example/drawingactivity/DrawActivity.kt:35]: Talked this over

File: app/src/main/java/com/example/drawingactivity/DrawActivity.kt (hunk 2 above)
Comment (brosscode) [reply app/src/main/java/com/example/drawingactivity/DrawActivity.kt:41]: I have learned from my mistakes

**Pull request merged**